### ISSUE TITLE
tsc doesn't catch the const variable reassign error in js file with --allowJs

### ISSUE THREAD

**Issue by Teamop** [Microsoft/TypeScript#26630]
When reassigning `const` variable in a .ts file, get error like `error TS2540: Cannot assign to 'a' because it is a constant or a read-only property.` But if in a .js file with `--allowJs`, got no error.

**TypeScript Version:**  3.0.1

**Search Terms:** const, allowJs

**Code**
```javascript
 const a = 1;
 a = 2;
```

**Expected behavior:**
Get error like in TS: `error TS2540: Cannot assign to 'a' because it is a constant or a read-only property.`
**Actual behavior:**
No error

**Playground Link:** 
N/A
**Related Issues:** 
N/A

**Comment by mattmccutchen** [comment]
You need to enable `--checkJs` if you want errors reported in JavaScript files.

**Comment by Teamop** [comment]
thanks @mattmccutchen
